src/mono/mono/mini/mini-s390x.c

```diff
@@ -2177,6 +2194,332 @@ mono_arch_peephole_pass_2 (MonoCompile *cfg, MonoBasicBlock *bb)
+                return OP_VREPIG;
+            default:
+                g_assert_not_reached ();
+                return -1;
+        }
+}
+
+static int
+simd_type_to_comp_op (int t)
+{
+        switch (t) {
+            case MONO_TYPE_I1:
+            case MONO_TYPE_U1:
+                return OP_VCEQBS;
+            case MONO_TYPE_I2:
+            case MONO_TYPE_U2:
+                return OP_VCEQHS;
+            case MONO_TYPE_I4:
+            case MONO_TYPE_U4:
+                return OP_VCEQFS;
+            case MONO_TYPE_I:
+            case MONO_TYPE_U:
+            case MONO_TYPE_I8:
+            case MONO_TYPE_U8:
+                return OP_VCEQGS;
+            case MONO_TYPE_R4:
+                return OP_VFCESBS;
+            case MONO_TYPE_R8:
+                return OP_VFCEDBS;
+            default:
+                g_assert_not_reached ();
+                return -1;
+        }
+}
+
+static int
+simd_type_to_gt_op (int t)
+{
+        switch (t) {
+            case MONO_TYPE_I1:
+                return OP_VCHBS;
+            case MONO_TYPE_U1:
+                return OP_VCHLBS;
+            case MONO_TYPE_I2:
+                return OP_VCHHS;
+            case MONO_TYPE_U2:
+                return OP_VCHLHS;
+            case MONO_TYPE_I4:
+                return OP_VCHFS;
+            case MONO_TYPE_U4:
+                return OP_VCHLFS;
+            case MONO_TYPE_I:
+            case MONO_TYPE_I8:
+                return OP_VCHGS;
+            case MONO_TYPE_U:
+            case MONO_TYPE_U8:
+                return OP_VCHLGS;
+            case MONO_TYPE_R4:
+                return OP_VFCHSBS;
+            case MONO_TYPE_R8:
+                return OP_VFCHDBS;
+            default:
+                g_assert_not_reached ();
+                return -1;
+        }
+}
+
+static int
+simd_type_to_extract_op (int t, int q)
+{
+    switch (t){
+        case SIMD_EXTR_ARE_ALL_SET:{
+            switch (q){
+                case CMP_LT:
+                case CMP_GT:
+                case CMP_GT_UN:
+                case CMP_LT_UN:
+                case CMP_EQ:
+                    return OP_CEQ;
+                case CMP_GE:
+                case CMP_LE:
+                case CMP_GE_UN:
+                case CMP_LE_UN:
+                    return OP_ICNLE; //ICNGE
+                default:
+                    g_assert_not_reached();
+                    return -1;
+            }
+        }
+        case SIMD_EXTR_IS_ANY_SET:{
+            switch (q){
+                case CMP_GT:
+                case CMP_LT:
+                case CMP_GT_UN:
+                case CMP_LT_UN:
+                case CMP_EQ:
+                    return OP_ICLE;
+                case CMP_GE:
+                case CMP_LE:
+                case CMP_GE_UN:
+                case CMP_LE_UN:
+                    return OP_ICNEQ;
+                default:
+                    g_assert_not_reached();
+                    return -1;
+            }
+        }
```

Comment:
Note: as you rework floating-point comparisons, this likely will also have to be updated.